resolve loader config for Hybridapp in 0.5.x

isao opened the pull request:
- use 0.5.x yui.config.x instead of older yui.x
- use relatve config.base and config.group.app.base
- rm obsolete, add some empty properties to override new defaults
- use yui.config.seed to point to resolved synthetic files, per @caridy. this avoids having to create a symlink to emulate the «static-prefix»/yui/ uri

only changes are to archetype/hybrid/app/application.json

Commit:
yui configs now in yui.config; rm dependencyCalc

Commit:
add groups

Commit:
revert base, whsp

Commit:
use symlinked yui/

Commit:
resolve loader config for Hybridapp in 0.5.x

config changes that avoid find/replace hacks, thanks Caridy

* use 0.5.x yui.config.x instead of older yui.x
* use relatve config.base and config.group.app.base
* rm obsolete, add some empty properties to override new defaults
* use yui.config.seed to point to resolved synthetic files, per @caridy. this avoids having to create a symlink to emulate the «static-prefix»/yui/ uri

caridy commented on lib/app/archetypes/app/hybrid/application.json.hb:
```diff
@@ -25,15 +21,30 @@
+                ],
+                "base": "../yahoo.libs.yui/",
+                "combine": false,
+                "comboBase": "",
```
remove comboBase, not needed.

caridy commented on lib/app/archetypes/app/hybrid/application.json.hb:
```diff
@@ -25,15 +21,30 @@
+                "groups": {
+                    "app": {
+                        "combine": false,
+                        "comboBase": "",
```
remove comboBase, not needed.

caridy commented:
please, check the notes

caridy commented on lib/app/archetypes/app/hybrid/application.json.hb:
```diff
@@ -25,15 +21,30 @@
+                    "/yahoo.libs.yui/loader-base/loader-base-min.js",
+                    "loader-yui3-resolved{langPath}",
+                    "loader-app",
+                    "loader-app-base{langPath}"
```
use loader-app-resolved{langPath}, which behaves better, and in any case u are already using loader-yui3-resolved.

a quick note on this:

if you plan to run functional tests, it will not work because yui-test is not part of the resolved meta ;) for that u have to use "/yahoo.libs.yui/loader-yui3/loader-yui3-min.js" instead of "loader-yui3-resolved{langPath}"

Commit:
apply feedback from Caridy: "use loader-app
-resolved{langPath}, which behaves
better, and in any case u are already using 
loader-yui3-resolved."

Commit:
rm comboBase per Caridy feedback

caridy commented:
+1

Commit:
for subdir case, 
- build.hybridapp.forceRelativePaths: false
- yui.config.base: "/yahoo.libs.yui/"
- yui.config.groups.app.base: "/"

doesn't work via http, but should for crt://

caridy commented:
@isao I like this solution better (I mean not using forceRelativePaths).
